### DISCUSSION TITLE
A Beginner's Struggle with ROCKET Model Prediction

### DISCUSSION THREAD

**Discussion by 14294302** [timeseriesAI/tsai#834] [Q&A]
Dear author,

I am a newcomer to machine learning, and I followed the tutorial in 'tutorial_nbs/02_ROCKET_a_new_SOTA_classifier.ipynb' to run the ROCKET model, achieving excellent results. However, I am unsure of how to make predictions using 'learn' because the 'Xtrain' and 'Ytrain' in this tutorial seem different from my previous training data. Could you please provide more detailed instructions on how to use the ROCKET model for predicting 'Xtrain'? Thank you.

**Comment by MOREDataset** [comment]
I couldn't understand the problem but there are multiple tutorials about
how to use different models for prediction.

**Reply by 14294302** [reply]
In 02_ROCKET_a_new_SOTA_classifier I use the follow code in it 

f_mean = X_train.mean(axis=0, keepdims=True)
f_std = X_train.std(axis=0, keepdims=True) + eps
X_train_norm = (X_train - f_mean) / f_std
X_valid_norm = (X_valid - f_mean) / f_std
...
classifier.fit(X_train_norm, y_train)
preds = classifier.predict(X_valid_norm)

it tell me :
XGBoostError: [18:44:43] C:\buildkite-agent\builds\buildkite-windows-cpu-autoscaling-group-i-0fdc6d574b9c0d168-1\xgboost\xgboost-ci-windows\src\c_api\c_api_utils.h:167: Invalid missing value: null

then I use the code in previous tutorial:
probas, _, preds = learn.get_X_preds(X)

it tell me:
RuntimeError: mat1 and mat2 shapes cannot be multiplied (64x10005 and 20000x5)
